### PR TITLE
Fix: perfect dark mode for all platforms

### PR DESCRIPTION
**Pull request by shenlebantongying**

### CODE REVIEW

**Comment by shenlebantongying** [comment]
Linux/KDE:
Breeze Light
image

Breze dark
image

**Comment by shenlebantongying** [comment]
macOS 
Screenshot 2022-12-25 at 12 45 00 AM
Screenshot 2022-12-25 at 12 46 02 AM

**Comment by shenlebantongying** [comment]
Windows

image
image

**Comment by shenlebantongying** [comment]
This PR is ready. The dark theme now looks perfect on all platforms @xiaoyifang 

I set the group combobox to palatee(window) and it looks almost like before (the qlineargradient).

File: qt-style.css [hunk 1]
```diff
@@ -1,140 +1,107 @@
...
-  image: url(:/icons/closetab-hover.png)
-}
+GroupComboBox::drop-down {
+    subcontrol-origin: padding;
+    subcontrol-position: top right;
+    width: 15px;
+    border-left: 1px solid darkgray;
+    border-right: 1px transparent;
 
+    /* Hack: For unknown reason, if the padding right is unset, the drop-down won't show full content */
```
Comment (shenlebantongying) [inline qt-style.css:49]: To deal with this bug, the dropdown cannot be shown after using dark mode.

https://forum.freemdict.com/t/topic/11495/1883

I think there are some mixing problems when using CSS, but that probably requires us to get rid of the CSS.

File: mainwindow.cc [hunk 2]
```diff
@@ -445,7 +445,6 @@ MainWindow::MainWindow( Config::Class & cfg_ ):
   closeCurrentTabAction.setShortcutContext( Qt::WidgetWithChildrenShortcut );
   closeCurrentTabAction.setShortcut( QKeySequence( "Ctrl+W" ) );
   closeCurrentTabAction.setText( tr("Close current tab") );
-  closeCurrentTabAction.setIcon( QIcon(":/icons/closetab-hover.png") );
```
Comment (xiaoyifang) [inline mainwindow.cc:448]: can this line left for windows only.I think it looks better than default.

Comment (shenlebantongying) [reply mainwindow.cc:448]: I just forced push the change. I added windows specific CSS code to custom close buttons.

File: mainwindow.cc [hunk 3]
```diff
@@ -1226,13 +1225,21 @@ void MainWindow::applyQtStyleSheet( QString const & displayStyle, QString const
...
 #endif
 
-  if ( displayStyle.size() )
-  {
-    // Load an additional stylesheet
-    QFile builtInCssFile( QString( ":/qt-style-st-%1.css" ).arg( displayStyle ) );
-    if ( builtInCssFile.open( QFile::ReadOnly ) )
-      css += builtInCssFile.readAll();
-  }
+#if defined(Q_OS_WIN)
```
Comment (xiaoyifang) [inline mainwindow.cc:1228]: I think ,this section can put into a seperator qt css file. such as qt-style-windows.
and load this file when in Window platform

Comment (shenlebantongying) [reply mainwindow.cc:1228]: Sure. I also keep the `qt-style-mac.css` there in case we need adjustments later.

Comment (shenlebantongying) [reply mainwindow.cc:1228]: I forced push the change

**Comment by xiaoyifang** [comment]
https://github.com/goldendict/goldendict/issues/719

**Comment by vitaly-zdanevich** [comment]
How to enable this, on AppImage?

**Pull request merged**